### PR TITLE
Feature/backend update delete

### PR DESCRIPTION
**Pull request by predrag-babic**

### CODE REVIEW

File: backend-update-and-delete_products/components/Product/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!data) {
+    return;
+  }
+
+  async function handleEditProduct(event) {
+    const formData = new FormData(event.target);
```
Comment (F-Kirchhoff) [inline backend-update-and-delete_products/components/Product/index.js:27]: you don't prevent the default

File: backend-update-and-delete_products/components/Product/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    mutate();
+    setIsEditMode(false);
+  }
+
+  async function handleDeleteProduct() {
+    const response = await fetch(`/api/products/${id}`, {
+      method: "DELETE",
+    });
+    if (response.ok) {
+      await response.json();
```
Comment (F-Kirchhoff) [inline backend-update-and-delete_products/components/Product/index.js]: you don't do anything with the response data, you can remove this line.

File: backend-update-and-delete_products/pages/api/products/[id].js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  if (request.method === "PUT") {
+    const updatedProduct = request.body;
+    await Product.findByIdAndUpdate(id, updatedProduct);
+    response.status(200).json({ message: "Product successfully updated" });
+    return;
+  }
+
+  if (request.method === "DELETE") {
+    await Product.findByIdAndDelete(id);
```
Comment (F-Kirchhoff) [inline backend-update-and-delete_products/pages/api/products/[id].js:26]: what about the reviews connected to this product? They are not deleted yet.

**Review by F-Kirchhoff — COMMENTED** [review]
Nicely done! I only spotted a hidden bug, thats quite advanced to find.

File: backend-update-and-delete_products/components/Product/index.js (hunk 1 above)
Comment (predrag-babic) [reply backend-update-and-delete_products/components/Product/index.js:27]: Thanks, i´ve updated this one.

File: backend-update-and-delete_products/pages/api/products/[id].js (hunk 3 above)
Comment (predrag-babic) [reply backend-update-and-delete_products/pages/api/products/[id].js:26]: I am not sure what do you mean with reviews... When i delete the product, i can´t see reviews anymore.. 
If you mean the reviews and comments that i build in submit form, they are not in this PR and project.

**Pull request merged**